### DISCUSSION TITLE
TOC for the output

### DISCUSSION THREAD

**Discussion by adrianboangiu** [schemacrawler/SchemaCrawler#1354] [Q&A]
Hello,

I have a database of about 900 tables and about the same number of views. The html file size for this database (in standard info-level) is 15MB. Therefore, it is very difficult to "navigate" inside. I am aware about the possibility to filter the content but this option does not fit what I try to achieve with schemacrawler.
I wonder if there is any possibility to obtain a TOC (table of contents) section in case the output supports this kind of feature (html, pdf). I notice that the foreign keys descriptions implement links to the referred columns. So, I was thinking that a TOC section can contain links to all the tables in the output.

Thank you

**Comment by sualeh** [comment]
@adrianboangiu Since each use case requires a different view, SchemaCrawler allows you to script exactly what you would like to see for schema output, using a language of your choice. Just like the default HTML and text outputs are built-in, SchemaCrawler also comes with Python scripts that can produce Mermaid, PlantUML or DBML output. You can use a templating language like Mustache or Freemarker, or a scripting language like Python or Ruby to create your own customized output.

Take a look at [Automatically Document Your Database in Markdown](https://dev.to/sualeh/automatically-document-your-database-in-markdown-elf) to get started towards creating your own TOC and HTML output.
